src/lib/hooks/swap/useConvenienceFee.ts

```diff
@@ -0,0 +1,43 @@
+import { Percent } from '@uniswap/sdk-core'
+import useActiveWeb3React from 'hooks/useActiveWeb3React'
+import { useAtom } from 'jotai'
+import { feeOptionsAtom } from 'lib/state/swap'
+import { useCallback, useEffect } from 'react'
+
+interface FeeOptionsArgs {
+  convenienceFee?: number
+  convenienceFeeRecipient?: string | string | { [chainId: number]: string }
+}
+
+export default function useConvenienceFee({ convenienceFee, convenienceFeeRecipient }: FeeOptionsArgs) {
+  const { chainId } = useActiveWeb3React()
+  const [feeOptions, updateFeeOptions] = useAtom(feeOptionsAtom)
+
+  const setToDefaults = useCallback(() => {
```

Comment:
nit: If you import a default value this becomes a one-liner, and I think you should omit the callback and just use it in the `useEffect` directly.